public/pages/ComposableTemplates/containers/ComposableTemplates/ComposableTemplates.tsx

```diff
@@ -89,13 +95,33 @@ class ComposableTemplates extends Component<ComposableTemplatesProps, Composable
       composableTemplates: [],
       loading: false,
       selectedTypes: [],
+      dataSourceId: props.dataSourceId,
+      dataSourceLabel: props.dataSourceLabel,
     };
 
     this.getComposableTemplates = debounce(this.getComposableTemplatesOriginal, 500, { leading: true });
   }
 
   getComposableTemplates: () => Promise<void> | undefined;
 
+  static getDerivedStateFromProps(nextProps: ComposableTemplatesProps, prevState: ComposableTemplatesState) {
+    if (nextProps.dataSourceId != prevState.dataSourceId || nextProps.dataSourceLabel != prevState.dataSourceLabel) {
+      return {
+        dataSourceId: nextProps.dataSourceId,
+        dataSourceLabel: nextProps.dataSourceLabel,
+      };
+    }
+    return null;
+  }
```

Comment:
It's better that we can declare a MDSEnableComponent, which extends React.Component and with the `getDerivedStateFromProps` method, as I see copy / paste across multiple components.

Reply:
Ack, addressed as part of next commit.